view/frontend/templates/checkout/minicart_addons.phtml

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2020 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/** @var \Magento\Framework\View\Element\Template $block */
+/** @var \Bolt\Boltpay\ViewModel\MinicartAddons $miniCartView */
+$miniCartView = $block->getViewModel();
+if (!$miniCartView->shouldShow()) return;
+?>
+<script>
+    require([
+        'jquery', 'Magento_Customer/js/model/customer',
+    ], function ($, customer) {
+        function append_minicart_addons () {
+            require(['uiLayout'], function (layout) {
+                layout(<?php echo $miniCartView->getLayoutJSON(); ?>);
+
+                <?php if($miniCartView->configHelper->displayRewardPointsInMinicartConfig()):?>
+                $('#minicart-content-wrapper').on('click', '.totals.rewardpoints .action.delete', function (e) {
+                    e.preventDefault();
+                    $.ajax({
+                        url:        window.checkoutConfig.review.reward.removeUrl,
+                        type:       'GET',
+                        cache:      false,
+                        dataType:   'json',
+                        showLoader: true,
+                        success:    function (response) {
+                            $(document).trigger('bolt:createOrder');
+                            require(
+                                [
+                                    'Magento_Checkout/js/action/get-payment-information',
+                                    'Magento_Customer/js/customer-data'
+                                ],
+                                function (getPaymentInformationAction, customerData) {
+                                    getPaymentInformationAction().always(function () {
+                                        customerData.reload(['messages'], true);
+                                    });
+                                }
+                            );
+                        }
+                    });
+                });
+                <?php endif; ?>
+            });
+        }
+
+        if (typeof window.checkoutConfig == 'undefined') {
+            $.post('<?php echo $block->getUrl('boltpay/cart/checkoutconfig'); ?>').success(function (result) {
```

Comment:
Could we use get instead of post? Post triggers all magento2 sections reload.

Reply:
@vitaliyreznikov `POST` was used to prevent JQuery from caching the request.  But, yes, we can and will use `GET` with the extra Jquery parameter `cacheable:false` to achieve this goal.   

This will slightly optimize the code because not ALL sections reload with `POST` but rather ONLY the messages section; And you are right, this minor load is not needed for this particular request.

Reply:
I don't think this necessary, but magento2 code has
```
    $.ajaxSetup({
        cache: false
    });
```

So looks like we can skip GET parameters, just to be short.
But this is up to you

Reply:
yep. I discovered this and commented on this yesterday in the comments below. 
https://github.com/BoltApp/bolt-magento2/pull/725#discussion_r437519342

Let's opt for the tried and true (@danac-gs calls it safer) approach that makes sure that if` `$.ajaxSetup` was called from elsewhere with conflicting configs, it does not affect our code.